server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/models/VisualizationResource.java

```diff
@@ -68,10 +68,25 @@ public void addLayer(Layer layer) {
         }
     }
 
-    public void removeLayer(Layer layer) {
+    public void updateLayer(Layer layer) {
+        boolean updated = false;
+        // if it exist, replace the existing layer with new one same order
+        for (int i = 0; i < layers.size(); i++) {
+            if (layers.get(i).getLayerId().equals(layer.getLayerId())) {
+                layers.set(i, layer);
+                updated = true;
+                break;
+            }
+        }
+       if (!updated) {
+            throw new IllegalArgumentException("Layer with ID " + layer.getLayerId() + " does not exist.");
+        }
+    }
+
+    public void removeLayerbyId(String layerId) {
```

Comment:
minor nitpick, should this be removeLayerById?

Reply:
Done